NitroxClient/MonoBehaviours/DiscordRP/AcceptRequest.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections;
+using NitroxClient.MonoBehaviours.DiscordRP;
+using UnityEngine;
+
+namespace NitroxClient.MonoBehaviours.Gui.MainMenu
+{
+    public class AcceptRequest : MonoBehaviour
+    {
+        public DiscordController DiscordRp;
+        public DiscordRpc.DiscordUser Request;
+
+        Rect discordWindowRect = new Rect(Screen.width / 2 - 250, 100, 500, 250);
+        Texture avatar;
+
+        public void Awake()
+        {
+            DontDestroyOnLoad(gameObject);
+        }
+
+        public void Start()
+        {
+            DiscordRp = gameObject.GetComponent<DiscordController>();
+            StartCoroutine(LoadAvatar(Request.userId, Request.avatar));
+        }
+
+        public void OnGUI()
+        {
+            discordWindowRect = GUILayout.Window(0, discordWindowRect, DoDiscordWindow, "Server join request from Discord");
+        }
+
+        private void SetGUIStyle()
+        {
+            GUI.skin.textArea.fontSize = 18;
+            GUI.skin.textArea.richText = false;
+            GUI.skin.textArea.alignment = TextAnchor.MiddleLeft;
+            GUI.skin.textArea.wordWrap = true;
+            GUI.skin.textArea.stretchHeight = true;
+            GUI.skin.textArea.padding = new RectOffset(10, 10, 5, 5);
+
+            GUI.skin.label.fontSize = 14;
+            GUI.skin.label.alignment = TextAnchor.MiddleRight;
+            GUI.skin.label.stretchHeight = true;
+            GUI.skin.label.fixedWidth = 140;
+            GUI.skin.label.fixedHeight = 140;
+
+            GUI.skin.button.fontSize = 14;
+            GUI.skin.button.stretchHeight = true;
+        }
+
+        private void DoDiscordWindow(int windowId)
+        {
+            SetGUIStyle();
+            using (GUILayout.VerticalScope v = new GUILayout.VerticalScope("Box"))
+            {
+                using (GUILayout.HorizontalScope h = new GUILayout.HorizontalScope())
+                {
+                    GUILayout.Label(avatar);
+                    GUILayout.Label("Player " + Request.username + " wants to join your server.");
+                }
+                using (GUILayout.HorizontalScope b = new GUILayout.HorizontalScope())
+                {
+                    if (GUILayout.Button("Accept"))
+                    {
+                        CloseWindow(1);
+                    }
+
+                    if (GUILayout.Button("Deny"))
+                    {
+                        CloseWindow(0);
+                    }
+                }
+            }
+        }
+
+        private void CloseWindow(int respond)
```

Comment:
Right now the request window will stay open forever as far as I've seen. Can you add a close button to the UI and make it auto close when the request token is expired?

You can start a coroutine and just `yield return new WaitForSeconds(x)` where x is the time until the token expires. Or just enter a suffient but short amount.